### DISCUSSION TITLE
How can I open a Pull Request after pushing a branch?

### DISCUSSION THREAD

**Discussion by Mo0rBy** [NeogitOrg/neogit#1205] [Q&A] [answered]
I noticed that this section existed within the default configuration:
```lua
  git_services = {
    ["github.com"] = "https://github.com/${owner}/${repository}/compare/${branch_name}?expand=1",
    ["bitbucket.org"] = "https://bitbucket.org/${owner}/${repository}/pull-requests/new?source=${branch_name}&t=1",
    ["gitlab.com"] = "https://gitlab.com/${owner}/${repository}/merge_requests/new?merge_request[source_branch]=${branch_name}",
  },
```

This made me realise that I should be able to open a PR for my branch right after pushing. However, I cannot find how to do it. There is no obvious keymapping for this action within the default configuration. I have looked at the help docs and they don't give much more info > https://github.com/NeogitOrg/neogit/blob/master/doc/neogit.txt#L627

I can't see any obvious option within the Branch popup either.
```
Variables
d branch.adding-neogit-nvim-plugin.description unset
u branch.adding-neogit-nvim-plugin.merge unset
  branch.adding-neogit-nvim-plugin.remote unset
R branch.adding-neogit-nvim-plugin.rebase [true|false|pull.rebase:false]
p branch.adding-neogit-nvim-plugin.pushRemote [origin]

Arguments
-r Recurse submodules when checking out an existing branch (--recurse-submodules)

Checkout                                Create           Do            
b branch/revision      c new branch     n new branch     C Configure...
l local branch         s new spin-off   S new spin-out   m rename      
r recent branch        w new worktree   W new worktree   X reset       
                                                         D delete  
```

Could anyone tell me where this is? It must be somewhere and I'm just being blind.

**Comment by CKolkey** [comment, marked as answer]
It requires the branch has a configured upstream - this is set up automatically when you push there - `Pu`. Note that a pushRemote will _not_ enable this, just an upstream... because... semantics, I guess.

Once thats there, on the branch popup the `o` action becomes available.

I'll make the UI clearer what needs to happen there.

**Reply by Mo0rBy** [reply]
Ah nice thanks! Just an FYI, this feature requires Neovim v0.10.0

**Reply by CKolkey** [reply]
Thats right - `vim.ui.open` only exists in 0.10
